src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -274,18 +266,22 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
     reshape_to_static(kvcache_model, 1u, m_kvcache_desc.total_size, axes);
 
     auto npudesc = extract_npu_descriptor(plugin);
-
     ov::AnyMap properties_copy = std::move(other_props);
-    auto prefill_config = get_default_prefill_config(model, npudesc);
-    // NB: GENERATE_HINT is only applicable for default generate config!
+
+    auto prefill_config =
+        prefill_config_opt.value_or(get_default_prefill_config(prefill_model, npudesc)).as<ov::AnyMap>();
+
     const ::intel_npu::npuw::llm::GenerateHint generate_hint = m_cfg.get<::intel_npu::NPUW_LLM_GENERATE_HINT>();
```

Comment:
I'd assume it initially extracted from `npuw_llm_props`

Reply:
If `npuw_llm_props` has it, it will extracted from here or default value will be taken